spreadsheets.js

```diff
@@ -206,38 +214,93 @@ function find_pairing(service_account_auth, spreadsheet_key, white, black, callb
             }
             // Query for the appropriate row.
             // Again, this ought to work for both tournaments
-            pairings_sheet.getRows({
-                offset: 1,
-                limit: 100
-            }, function(err, rows) {
-                if (err) { return callback(err, rows); }
-                var potential_rows = [];
-                rows.forEach(function(row) {
-                    var row_black = row.black.toLowerCase();
-                    var row_white = row.white.toLowerCase();
-                    if (
-                        (row_black.indexOf(black) != -1 && row_white.indexOf(white) != -1)
-                    ) {
-                        potential_rows.push([row, false]);
-                    } else if (
-                        (row_white.indexOf(black) != -1 && row_black.indexOf(white) != -1)
-                    ) {
-                        potential_rows.push([row, true]);
-                    }
-                });
-                // Only update it if we found an exact match
-                if (potential_rows.length > 1) {
-                    return callback("Unable to find pairing. More than one row was returned!");
-                } else if(potential_rows.length < 1) {
-                    return callback("Unable to find pairing. No rows were returned!");
+            options = _.clone(options);
+            var defaults = {
```

Comment:
I have some constants that I intend to add to the config. The could be used here to make this more flexible.

I'll get to that later. This is fine for the rush we are in to get Chesster back up.